### PR TITLE
Removing Deprecated Fields from Several Objects

### PR DESCRIPTION
**Pull request by mitchute**
Addresses #4664 and #5600. The following unused fields have been removed:

HVACTemplate:System:Unitary - Dehumidification Control Zone Name
HVACTemplate:System:UnitarySystem - Dehumidification Control Zone Name
Branch - Maximum Flow Rate
Branch - Component n Branch Control Type
ChillerHeater:Absorption:DirectFired - Chiller Flow Mode
SetpointManager:SingleZone:Humidity:Minimum - Control Variable
SetpointManager:SingleZone:Humidity:Maximum - Control Variable
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by mitchute** [comment]
@Myoldmopar @mjwitte @JasonGlazer 

In `HVACTemplate:System:Unitary` there is a field for "Dehumidification Control Zone Name." In the idd it says that the field is not currently used. Is that something we want to take out? Or do we leave it for later?

**Comment by mjwitte** [comment]
@mitchute Dehumidification Control Zone Name should be removed from both `HVACTemplate:System:Unitary` and `HVACTemplate:System:UnitarySystem`.  In both of these system types, the dehumidification control zone is the same as the Control Zone or Thermostat Location Name.  These changes will require updates to expandobjects, so you can make this a new issue or deal with it here.

**Comment by mjwitte** [comment]
@mitchute I suspect the integration test failures are due to the Branch object changes in CreateNewDualDuct (github won't display epfilter because it's too big . . . so pasting in some code here.  Here's an example from line 21194 in current develop epfliter.f90 (old):

```
  CALL AddToObjFld('Component Outlet Node Name', base + ddsAirHandlerNameOff,' Mixed Air Outlet')
  CALL AddToObjStr('Component Branch Control Type','Passive',(supFanKind .EQ. sfkNone))
```

The boolean expression at the end the old Branch Control Type line determines if this is the end of the object.  This needs to be added to the line above.  There are quite a few similar places like this in CreateNewDualDuct.

Also, this block starting at line 28624 in this branch (new code), the last line of the first IF block also needs a `.TRUE.`, because the old Branch Control Type line was outside the IF/ELSEIF:

```
    IF (loopPumpConfig .EQ. pumpConstant) THEN
      CALL AddToObjFld('Component Name', base + cwpNameOff,' ChW Supply Pump')
      CALL AddToObjFld('Component Inlet Node Name', base + cwpNameOff,' ChW Supply Inlet')
      CALL AddToObjFld('Component Outlet Node Name', base + cwpNameOff,' ChW Pump Outlet')
    ELSEIF (loopPumpConfig .EQ. pumpConstPriVarSec) THEN
      CALL AddToObjFld('Name', base + cwpNameOff,' ChW Primary Supply Pump')
      CALL AddToObjFld('Component Inlet Node Name', base + cwpNameOff,' ChW Supply Inlet')
      CALL AddToObjFld('Component Outlet Node Name', base + cwpNameOff,' ChW Primary Pump Outlet',.TRUE.)
    END IF
```

Otherwise, the (many) epfilter changes look correct.

@Myoldmopar Touching the Branch object here impacts any idf file with an air loop or plant loop.  In #5582 I hope to remove the Branch Maximum Flow Rate field as well.  Just want to be sure we're ready to take this plunge.  And there's been suggestions to remove the component input/output node names as well (or reduce their redundancy in this object).  That's a larger discussion but would be good to make that decision now so there's only one big transition to Branch all in the same release.

**Comment by mjwitte** [comment]
@Myoldmopar @lefticus Looking at the CI test results comments above, the new summaries of diffs are great.  Would be nice to add a statement of how many tests failed to complete so it's more obvious that failures are due to fatal errors or crashes rather compared to failures due to big diffs.

**Comment by mitchute** [comment]
@mjwitte thanks for looking at that. I believe it is correct now.

**Comment by Myoldmopar** [comment]
@mitchute Can you confirm that the branch control type isn't being used anywhere on the air-side?  @EnergyArchmage feels like it may have been.  Just wanting to confirm.

**Comment by mitchute** [comment]
@Myoldmopar I have not seen where it is still being used, but I'm still working through errors so I can't say for 100% certain. From my [last commit](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-1ef8a4235ee33a657dbd26b5428c05adefb113e2-i386-Windows-7-VisualStudio-12.html) it looks like there are (at least) two remaining issues: one related to ExpandObjects, and the other is a branch input failure during sizing. If I had to guess, I would say that the second may be related to what @EnergyArchmage is saying, but I need to get into it and figure it out.

**Comment by mitchute** [comment]
@Myoldmopar @mjwitte @JasonGlazer @EnergyArchmage anyone else...?

CI is coming back green and I think this is ready for review. There are a lot of changes so you will need to diff the repo offline. AUD and BND diffs are expected. Let me know if I missed anything.

### Summary

#### Branch

Removed: _Control Type_ field

Core code from GetBranchInput was moved to GetSingleBranchInput for unit testing.

Transition was tested and added for V8.6.0.

#### SetPointManager:SingleZone:Humidity:Min/Max

Removed: _Control Variable_ field
Removed: _Schedule Name_ field

Transition was tested and added for V8.6.0.

#### ChillerHeater:Absorption:DirectFired

Removed: _Chiller Flow Mode_ field

Transition was tested and added for V8.6.0.

#### HVACTemplate:Summary:Unitary

Removed: _Dehumidification Control Zone Name_ field

Transition was tested and added for V8.6.0.

#### HVACTemplate:Summary:UnitarySystem

Removed: _Dehumidification Control Zone Name_ field

Transition was tested and added for V8.6.0.

#### SurfaceConvectionAlgorithm:Outside

Code was removed to handle "Simple," "Detailed," "BLAST," and "DOE2" options

Transition was already in place from V6.0.0.

#### SurfaceConvectionAlgorithm:Inside

Code was removed to handle "Simple" option

Transition was already in place from V6.0.0.

#### Zone

Code was removed to handle "Detailed" option from _Inside Convection Algorithm_ field
Code was removed to handle "Simple," "Detailed," and "BLAST" option from _Outside Convection Algorithm_ field

Transition was already in place from V6.0.0.

#### Exterior:FuelEquipment

Code was removed to handle "Gas," and "LPG" options from _Fuel Use Type_ field

Transition was tested and added for V8.6.0.

The report variable still reports "Gas" instead of "NaturalGas." Let me know if you want this changed to be consistent.

#### Controller:MechancialVentilation

Code was removed to handle "VRP," and "IAQP" options from _System Outdoor Air Method_ field

Transition was tested and added for V8.6.0

#### Condenser/PlantEquipmentOperationSchemes

Code was removed to handle "Load Range Based Operation" option from _Control Scheme_ field

Transition was already in place from V3.0.0. 

The only place I ran into this was in the performance tests, but they are all commented out currently and are not running. I'm guessing that will be a future task for someone separate from this.

### To do:
- Address review comments
- Fix unused field warning
- Change "Gas" to "NaturalGas" in Exterior:FuelEquipment?

**Comment by Myoldmopar** [comment]
@mitchute @mjwitte I haven't looked at the diffs yet (in a while anyway), but before I do, were there ExpandObjects changes for any of these fields?  And if so, are they done already?  That always seems to pop up right at the end of these types of changes.

**Comment by mjwitte** [comment]
@Myoldmopar Yes, there are expandobject changes, some of which were causing problems.  I'm assuming @mitchute has cleaned those up, but these we hard to spot because the files simply failed - which isn't itemized in the CI report (hint-hint would be fantastic to catch those and have a message that says " 6 runs crashed" " 5 runs had fatal errors" (I know - some of the runs a built to have fatal errors, so . . . )  All tests are passing, so these must have been fixed.
But I see [build warnings.](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-4cd6803c8366a53fb042bf69d8d4f1c5f0086afa-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html)

**Comment by mitchute** [comment]
@Myoldmopar @mjwitte Yes, there were. Mike commented part way through and I cleaned them up.

**Comment by mitchute** [comment]
And yes, there is one build warning. I figured there would be comments to address so I will clean that up when you all get through with the review.

**Comment by Myoldmopar** [comment]
@mitchute Can you resolve the merge conflict?  They are just in the BranchInputManager, and the accompanying unit test.  I think these were semi-expected because you and @mjwitte were both touching the branch objects?  Anyway, as soon as you do that, I'll generate and post a cleaned up diff between this branch and develop, highlighting only the code changes and not all the idfs.

**Comment by mitchute** [comment]
@Myoldmopar that should do it. Unit tests run fine over here. We'll see what CI says.

File: src/EnergyPlus/BranchInputManager.cc [hunk 1]
```diff
@@ -1518,7 +1518,7 @@ namespace BranchInputManager {
 		int NumNodes; // Number of Nodes from NodeInputManager
 
 		Branch( BCount ).Name = Alphas( 1 );
-		Branch( BCount ).MaxFlowRate = Numbers( 1 );
+//		Branch( BCount ).MaxFlowRate = Numbers( 1 );
```
Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc]: @mjwitte @Myoldmopar Do you want this to come out like I did with the other deprecated Branch fields on this PR?

Comment (Myoldmopar) [inline src/EnergyPlus/BranchInputManager.cc]: We mentioned this on one of our sizing calls and I can't remember what we resolved.  I think we decided that yes, we'd like to go ahead and get rid of it.  If you can add it here, that's cool, but if you need/want to put it in a separate, smaller branch of changes, that's fine with me too.

Comment (mitchute) [inline src/EnergyPlus/BranchInputManager.cc]: @Myoldmopar I went ahead and added it. What's another 5k diffs for this PR!?!?

Transition and ExpandObjects was updated. I can clean up any remaining issues after CI reports back.

**Comment by Myoldmopar** [comment]
The cleaned up diff is posted here: http://myoldmopar.github.io/quick-diff-spot/matts_diff_cleaned.html

It's not as nice as Github's diff engine of course, but it does show the changes without any idf or imf changes getting in the way.  If we wanted to do this on Github, we could potentially create a new branch off of this one, but revert away the idf/imf changes.  Probably not worth the overhead of doing that.

For reference, the way I generated the diff was:

```
$ git diff --color develop..DepricatedFields `git diff develop..DepricatedFields --name-status | grep -v '.idf' | grep -v '.imf' | cut -c 3-` > /tmp/diffs_file.diff
$ ansifilter -l -i /tmp/diffs_file.diff -H -o /tmp/diffs_file.html
```

**Comment by mitchute** [comment]
@Myoldmopar The latest commit should cleanup the CI failures. The rest of the "failures" and diffs are expected. Removing an autosize field shows up as a large table diff.

**Comment by nrel-bot-3** [comment]
@mitchute @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mitchute @lgentile it has been 66 days since this pull request was last updated.

**Comment by mitchute** [comment]
@Myoldmopar I merged develop in one more time. What CI is reporting all seems OK. Two failures were due to new input files that had not been transitioned. The "Big Diffs" all appear to be due to string diffs, or changes in the number of input fields. Let me know if you see anything you want me to look at.

**Comment by Myoldmopar** [comment]
@mitchute It looks like `AirflowNetwork_PressureControl` still [fails](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-e28a8ba8b4355f3e5d1eaf4b5d07a1887f7284cc-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html).  Can you have a look there?

**Comment by mitchute** [comment]
@Myoldmopar that should clean it up. I forgot that we pulled the max flow rate field from the branch objects.

**Comment by Myoldmopar** [comment]
@mitchute Looking through the results, I see some 1 eso big diff, and some bnd diffs.  For example the [`DOAToFanCoilInlet`](http://energyplus.s3-website-us-east-1.amazonaws.com/regressions/2016-08/33c0bc6091be818f666031b95819d187fdc3323a-46280424e5d5403e9050e54701adf10e3238b349/DOAToFanCoilInlet/x86_64-MacOS-10.9-clang/eplusout.bnd.diff.html) file is showing non-parent-node connection changes.  And [`DOASDualDuctSchool`](http://energyplus.s3-website-us-east-1.amazonaws.com/regressions/2016-08/33c0bc6091be818f666031b95819d187fdc3323a-46280424e5d5403e9050e54701adf10e3238b349/DOASDualDuctSchool/x86_64-MacOS-10.9-clang/eplusout.err.diff.html) is showing new errors in the error file.  Could you take a look when you get a chance?

**Comment by mitchute** [comment]
@Myoldmopar the `DOASDualDuctSchool` error file issues look like they are due to being out of sync with develop. Let's see if this latest commit will clean that up. Regarding the outlet-inlet node changes, there was an issue that was corrected by this work relating to that. I will need to compare actual loop topology in the idf to this to verify.

**Comment by mjwitte** [comment]
@mitchute @Myoldmopar Would like to get this in next.  Are you working on the errors today?  If not, I'll take a crack.

**Comment by Myoldmopar** [comment]
Have at it; for starters, it is complaining about the version ID in two files, just go to the regression warnings and search for "err diffs".

**Comment by mitchute** [comment]
@Myoldmopar @mjwitte All diffs I saw two days ago were expected. Let me know if you find something, though.

**Comment by mjwitte** [comment]
The failures may be more due to develop than this branch.  Working on fixing some failing performance tests.  Also, puzzled why there are some files with bnd diffs.  The node type shouldn't have changed due to this.  But will revisit once develop fixed and then re-merged into here.  Working on that now.

**Comment by mitchute** [comment]
The bnd diffs are expected, at least for the heat pump water heater files. If you look at the actual loop topology, you should find that it is now reporting correctly. Again, let me know if it isn't right.

**Comment by mjwitte** [comment]
@mitchute Fascinating.  It's because the two heatpump water heater test files just happened to have a few branches without a `Component 1 Branch Control Type` field.  That caused the old code to think `Component 1 Outlet Node Name` wasn't the end of the branch and called it an Internal node instead of an Outlet.

**Comment by mitchute** [comment]
@mjwitte Merged in one more time. I'll keep an eye on the results.

**Comment by mjwitte** [comment]
@mitchute OK Thanks.  We'll need to merge develop in again after #5822 drops in.

**Comment by mitchute** [comment]
OK, can do. I will keep an eye out for when that drops.

**Comment by Myoldmopar** [comment]
OK, I merged develop.  The conflict was related to @mjwitte's IDD cleanups.  They were easy except there appeared to be a little funny business with the SteamEquipment object.  I think I got it in there right, and 5ZoneWaterSystems passes successfully locally, so I think we're good.  @mitchute you may diff the idd from develop and this branch locally to take a better look, but I'll likely merge this tonight/tomorrow and then see about #5584 .

**Comment by Myoldmopar** [comment]
Thanks @mitchute!  If this is clean, I'll drop it in.

**Comment by Myoldmopar** [comment]
And it's hard to notice, but in the Linux runs, there was also a failing test file in the performance test suite, so I cleaned that one up, hopefully reducing this to just regressions @mitchute

**Comment by mitchute** [comment]
Thanks for catching that one!

**Comment by Myoldmopar** [comment]
OK, so 2 files had err/eio diffs.  They were the same file, and only because somehow the version number got pushed back to 8.5.  Anyway, there were no other problems with those files, just the version number.  I fixed those and I'm merging this now.

**Pull request merged**